spec/spec_helper.rb

```diff
@@ -1 +1,23 @@
+if ENV['COVERAGE']
+  # In order to archive an expected coverage analysis we need to eager load
+  # all logstash code base, otherwise it will not get a good analysis.
+
+  require "pluginmanager/main"
+  Dir.glob("lib/logstash/patches/*") do |file|
+    require file
+  end
+
+  Dir.glob("lib/logstash/*.rb") do |file|
+    require file
+  end
+
+  Dir.glob("lib/logstash/util/*") do |file|
+    next if ["lib/logstash/util/prctl.rb"].include?(file)
```

Comment:
this should not be MRI specific but Linux specific so AFAIK it is still relevant.